### PR TITLE
Fix resource requirements for sidecar/pod tasks

### PR DESCRIPTION
**Pull request by jeevb**
# TL;DR
Fixes task node resource overrides via `.with_overrides` in flytekit not applying to resource requirements in sidecar/pod tasks. Also fixes k8s plugin default resource requests not applying to sidecar/pod tasks.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Merges task node resource requirements with resource requirements of primary container in sidecar/pod task, after applying defaults where necessary.

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/1323

### CODE REVIEW

File: go/tasks/plugins/k8s/sidecar/sidecar_test.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ const ResourceNvidiaGPU = "nvidia.com/gpu"
 
 var resourceRequirements = &v1.ResourceRequirements{
 	Limits: v1.ResourceList{
-		v1.ResourceCPU:     resource.MustParse("1024m"),
-		v1.ResourceStorage: resource.MustParse("100M"),
+		v1.ResourceCPU:              resource.MustParse("2048m"),
+		v1.ResourceEphemeralStorage: resource.MustParse("100M"),
```
Comment (katrogan) [inline go/tasks/plugins/k8s/sidecar/sidecar_test.go:39]: 😁

**Review by katrogan — APPROVED** [review]
thank you!!!

**Review by katrogan — APPROVED** [review]
